pkg/foundation/grpcutil/websocket.go

```diff
@@ -82,7 +84,10 @@ func newWebSocketProxy(handler http.Handler, logger log.CtxLogger) *webSocketPro
 		pongWait:   defaultPongWait,
 		pingPeriod: (defaultPongWait * 9) / 10,
 	}
-
+	go func() {
+		<-ctx.Done()
+		proxy.done <- struct{}{}
+	}()
```

Comment:
Doing it this way we will only notify one websocket connection that Conduit is shutting down. We need to close the channel to notify all of them.

And while we're at it, we could simply assign `ctx.Done()` to `done`, it's already a channel that will be closed when the context is closed. We don't need this goroutine then.

Reply:
Good callout!